File: problems/2708_maximum_strength_of_a_group/solution.py
from typing import List


class Solution:
    def maxStrength(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1: return nums[0]
        neg = sorted([x for x in nums if x < 0])
        pos = [x for x in nums if x > 0]
        if len(neg) % 2 == 1:
            neg = neg[:-1]
        if not neg and not pos:
            return 0
        result = 1
        for x in neg + pos:
            result *= x
        return result

File: problems/2708_maximum_strength_of_a_group/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    long long maxStrength(vector<int>& nums) {
        if (nums.size() == 1) return nums[0];
        vector<int> neg, pos;
        for (int n : nums) { if (n < 0) neg.push_back(n); else if (n > 0) pos.push_back(n); }
        sort(neg.begin(), neg.end());
        if (neg.size() % 2 == 1) neg.pop_back();
        if (neg.empty() && pos.empty()) return 0;
        long long res = 1;
        for (int n : neg) res *= n;
        for (int n : pos) res *= n;
        return res;
    }
};
